### ISSUE TITLE
GivenThatWeWantToStoreAProjectWithDependencies.compose_multifile has become flaky

### ISSUE THREAD

**Issue by nguerrera** [dotnet/sdk#2089]
I'm making it run Windows only for now.

**Comment by nguerrera** [comment]
I don't think the test or code underneath it has changed in a long time, but the flakiness is recent.

**Comment by dsplaisted** [comment]
`Project_reference_compat` tests are also flaky on Ubuntu.  They are failing with what look like weird characters added at the end of paths:

> Restoring packages for /mnt/j/workspace/dotnet_sdk/release_2.1.3xx/debug_ubuntu16.04_prtest/artifacts/Debug/tmp/Project_refer---917A5525/Dependency1/Dependency1.csproj...

> /mnt/j/workspace/dotnet_sdk/release_2.1.3xx/debug_ubuntu16.04_prtest/artifacts/.dotnet/2.1.300-preview3-008414/sdk/2.1.300-preview3-008414/NuGet.targets(114,5): error : Could not find a part of the path '/mnt/j/workspace/dotnet_sdk/release_2.1.3xx/debug_ubuntu16.04_prtest/artifacts/.nuget/packages/system.appcontext/4.3.0/ref/netstandard1.6/ja/�a'. [/mnt/j/workspace/dotnet_sdk/release_2.1.3xx/debug_ubuntu16.04_prtest/artifacts/Debug/tmp/Project_refer---917A5525/Referencer/Referencer.csproj]

An example failure is [here](https://ci.dot.net/job/dotnet_sdk/job/release_2.1.3xx/job/debug_ubuntu16.04_prtest/42/).  This may or may not be the same problem as the `compose_multifile` test which Nick disabled.

@JeremyKuhne Any idea if this is related to File IO changes in the runtime?

**Comment by nguerrera** [comment]
Update: it is failing on Windows suddenly all of the time too. This leads me to think that there is some condition that causes a machine to get into a state where it starts to fail. Maybe temp is getting full or something like that.

**Comment by nguerrera** [comment]
cc @livarcocc

**Comment by nguerrera** [comment]
Got a repro locally with a hint at the problem:

[Microsoft.NET.Publish.Tests_netcoreapp2.0_x64.log](https://github.com/dotnet/sdk/files/1873180/Microsoft.NET.Publish.Tests_netcoreapp2.0_x64.log)

<details>
<summary>Stack trace</summary>

```
The "ResolvePackageAssets" task failed unexpectedly. [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
System.IO.IOException: The process cannot access the file 'D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\obj\Debug\netcoreapp2.0\win7-x64\NewtonsoftMultipleVersions.assets.cache' because it is being used by another process. [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle) [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options) [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share) [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at System.IO.File.Open(String path, FileMode mode, FileAccess access, FileShare share) [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheWriter..ctor(ResolvePackageAssets task) in /_/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:line 455 [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheReader..ctor(ResolvePackageAssets task) in /_/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:line 332 [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at Microsoft.NET.Build.Tasks.ResolvePackageAssets.ReadItemGroups() in /_/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:line 214 [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at Microsoft.NET.Build.Tasks.ResolvePackageAssets.ExecuteCore() in /_/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:line 206 [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at Microsoft.NET.Build.Tasks.TaskBase.Execute() in /_/src/Tasks/Common/TaskBase.cs:line 20 [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\Src\sdk\artifacts\Debug\tmp\multifileTargetManifests\NewtonsoftMultipleVersions.xml]
```

</details>

Looks like something is building the same project in parallel with different global properties and same intermediate output directory.

**Comment by nguerrera** [comment]
Fixed by #2159
